internal/pkg/manifest/env.go

```diff
@@ -102,9 +102,56 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID      *string              `yaml:"id,omitempty"`
-	CIDR    *IPNet               `yaml:"cidr,omitempty"`
-	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
+	ID                  *string              `yaml:"id,omitempty"`
+	CIDR                *IPNet               `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
+}
+
+type securityGroupConfig struct {
+	Ingress []securityGroupRule `yaml:"ingress,omitempty"`
+	Egress  []securityGroupRule `yaml:"egress,omitempty"`
+}
+
+// IsEmpty returns true if there is no security group rule defined.
+func (cfg securityGroupConfig) IsEmpty() bool {
+	return len(cfg.Ingress) == 0 && len(cfg.Egress) == 0
+}
+
+type securityGroupRule struct {
+	CidrIP     string `yaml:"cidr"`
+	FromPort   int    `yaml:"from_port"`
+	IpProtocol string `yaml:"ip_protocol"`
+	ToPort     int    `yaml:"to_port"`
+}
+
+//IsValid checks for valid security group config.
+func (cfg securityGroupConfig) IsValid() bool {
```

Comment:
Two questions!
1. Why don't we inline these logic in `func (cfg securityGroupConfig) validate() error`?
2. Do you have an example of an invalid manifest, according to this method?

Reply:
1. I think your recommendation make sense to inline `IsValid` inside `validate()`
2. I have added couple of tests demonstrating the invalid input.